pkg/flow/internal/controller/loader.go

```diff
@@ -311,6 +315,22 @@ func (l *Loader) splitComponentBlocks(blocks []*ast.BlockStmt) (componentBlocks,
 	return componentBlocks, serviceBlocks
 }
 
+func (l *Loader) populateDeclareNodes(g *dag.Graph, declares []*Declare) diag.Diagnostics {
+	var diags diag.Diagnostics
+	for _, declare := range declares {
+		node := NewDeclareNode(declare)
+		if g.GetByID(node.NodeID()) != nil {
```

Comment:
In populateComponentNodes, we only create the node if it doesn't already exist in the graph; can we do the same thing here?